src/pages/workspace/distanceRates/PolicyDistanceRateEditPage.tsx

```diff
@@ -0,0 +1,201 @@
+import OfflineWithFeedback from '@components/OfflineWithFeedback';
+import ScreenWrapper from '@components/ScreenWrapper';
+import Switch from '@components/Switch';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import * as CurrencyUtils from '@libs/CurrencyUtils';
+import * as ErrorUtils from '@libs/ErrorUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@navigation/types';
+import AdminPolicyAccessOrNotFoundWrapper from '@pages/workspace/AdminPolicyAccessOrNotFoundWrapper';
+import FeatureEnabledAccessOrNotFoundWrapper from '@pages/workspace/FeatureEnabledAccessOrNotFoundWrapper';
+import PaidPolicyAccessOrNotFoundWrapper from '@pages/workspace/PaidPolicyAccessOrNotFoundWrapper';
+import * as Policy from '@userActions/Policy';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type * as OnyxTypes from '@src/types/onyx';
+import type {Rate} from '@src/types/onyx/Policy';
+import RateModal from './RateModal';
+
+type PolicyDistanceRateEditPageOnyxProps = {
+    /** Policy details */
+    policy: OnyxEntry<OnyxTypes.Policy>;
+};
+
+type PolicyDistanceRateEditPageProps = PolicyDistanceRateEditPageOnyxProps & StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.DISTANCE_RATE_EDIT>;
+
+function PolicyDistanceRateEditPage({policy, route}: PolicyDistanceRateEditPageProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const {windowWidth} = useWindowDimensions();
+    const [isWarningModalVisible, setIsWarningModalVisible] = useState(false);
+    const [isRateModalVisible, setIsRateModalVisible] = useState(false);
+    const [isDeleteModalVisible, setIsDeleteModalVisible] = useState(false);
+
+    const policyID = route.params.policyID;
+    const rateID = route.params.rateID;
+    const customUnits = policy?.customUnits ?? {};
+    const customUnit = customUnits[Object.keys(customUnits)[0]];
+    const rate = customUnit.rates[rateID];
+    const currency = rate.currency ?? CONST.CURRENCY.USD;
+    const canDeleteRate = Object.keys(customUnit.rates).length > 1;
+    const canDisableRate = Object.values(customUnit.rates).filter((r) => r.enabled).length > 1;
+    const errorFields = rate.errorFields;
+
+    const showRateModal = () => {
+        setIsRateModalVisible(true);
+    };
+
+    const hideRateModal = () => {
+        setIsRateModalVisible(false);
+    };
+
+    const updateRate = (values: FormOnyxValues<typeof ONYXKEYS.FORMS.POLICY_DISTANCE_RATE_EDIT_FORM>) => {
+        Policy.updatePolicyDistanceRateValue(policyID, customUnit, [{...rate, rate: Number(values.rate) * CONST.POLICY.CUSTOM_UNIT_RATE_BASE_OFFSET}]);
+        Keyboard.dismiss();
+        hideRateModal();
```

Comment:
```suggestion
        Navigation.dismissModal();
```

Reply:
Are you sure that this is the correct behaviour? User might want to do something else or edit it again. If we change it to dismissModal then we close the RHP completely after editing rate value

Reply:
I agree with @MrMuzyk, we should just go back to the edit page.

Reply:
there are some pages that do this but i agree this is fine. lets keep it as it is.